lib/chef/provisioning/transport/ssh.rb

```diff
@@ -107,7 +111,7 @@ def write_file(path, content)
         execute("mkdir -p #{File.dirname(path)}").error!
         if options[:prefix]
           # Make a tempfile on the other side, upload to that, and sudo mv / chown / etc.
-          remote_tempfile = "/tmp/#{File.basename(path)}.#{Random.rand(2**32)}"
+          remote_tempfile = "#{@scp_temp_dir}/#{File.basename(path)}.#{Random.rand(2**32)}"
```

Comment:
Actually, because we reference `remote_tempfile` multiple places and they all have the same implementation, it should just be extracted to a method.